### ISSUE TITLE
segfault in llc

### ISSUE THREAD

**Issue by tkoenig1** [bagel99/llvm-my66000#12]
Another error from GSL, this time with a segfault.

Error message:
```
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.      Program arguments: llc --disable-lsr --enable-predication --enable-predication2 --enable-carry-generation --early-carry-coalesce --enable-vvm -march=my66000 step_opt.bc
1.      Running pass 'Function Pass Manager' on module 'step_opt.bc'.
2.      Running pass 'My66000 DAG->DAG Pattern Instruction Selection' on function '@gsl_odeiv2_step_apply'
 #0 0x00000000026da3dc llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/tkoenig/bin/llc+0x26da3dc)
 #1 0x00000000026d82b4 llvm::sys::RunSignalHandlers() (/home/tkoenig/bin/llc+0x26d82b4)
 #2 0x00000000026d8413 SignalHandler(int) (/home/tkoenig/bin/llc+0x26d8413)
 #3 0x00007f3f79eb78c0 __restore_rt (/lib64/libpthread.so.0+0x168c0)
 #4 0x0000000002497a7c (anonymous namespace)::ScheduleDAGRRList::ReleasePredecessors(llvm::SUnit*) (/home/tkoenig/bin/llc+0x2497a7c)
 #5 0x00000000024982ea (anonymous namespace)::ScheduleDAGRRList::Schedule() (/home/tkoenig/bin/llc+0x24982ea)
 #6 0x000000000253b620 llvm::SelectionDAGISel::CodeGenAndEmitDAG() (/home/tkoenig/bin/llc+0x253b620)
 #7 0x000000000253ea3c llvm::SelectionDAGISel::SelectAllBasicBlocks(llvm::Function const&) (/home/tkoenig/bin/llc+0x253ea3c)
 #8 0x0000000002540c78 llvm::SelectionDAGISel::runOnMachineFunction(llvm::MachineFunction&) (.part.845) (/home/tkoenig/bin/llc+0x2540c78)
 #9 0x0000000001c2df38 llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/home/tkoenig/bin/llc+0x1c2df38)
#10 0x0000000001ff984f llvm::FPPassManager::runOnFunction(llvm::Function&) (/home/tkoenig/bin/llc+0x1ff984f)
#11 0x0000000001ff9f31 llvm::FPPassManager::runOnModule(llvm::Module&) (/home/tkoenig/bin/llc+0x1ff9f31)
#12 0x0000000001ff8c17 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/home/tkoenig/bin/llc+0x1ff8c17)
#13 0x000000000079742b compileModule(char**, llvm::LLVMContext&) (/home/tkoenig/bin/llc+0x79742b)
#14 0x00000000006d7872 main (/home/tkoenig/bin/llc+0x6d7872)
#15 0x00007f3f7891829d __libc_start_main (/lib64/libc.so.6+0x3529d)
#16 0x000000000078f42a _start /home/abuild/rpmbuild/BUILD/glibc-2.31/csu/../sysdeps/x86_64/start.S:122:0
/home/tkoenig/bin/my66000: Zeile 6: 11982 Speicherzugriffsfehler  (Speicherabzug geschrieben) llc --disable-lsr --enable-predication --enable-predication2 --enable-carry-generation --early-carry-coalesce --enable-vvm -march=my66000 $b.bc
```
[tst2.zip](https://github.com/bagel99/llvm-my66000/files/10117709/tst2.zip)

**Comment by bagel99** [comment]
Fixed in commit d030246b63d36a6fb9cf3b6f5436e179ce074411.

**Comment by bagel99** [comment]
Fixed in commit d030246b63d36a6fb9cf3b6f5436e179ce074411.
